### PR TITLE
Feat/pfm issue 10674 generate company wide release notes

### PR DESCRIPTION
**Pull request by stefancplace**
added generation of Release Notes that can be published to cplace.docs.io

### CODE REVIEW

**Comment by stefancplace** [comment]
to be tested:
-> generated changelog did not change
-> commands used for orbs still working

File: src/commands/release-notes/GenerateReleaseNotes.ts [hunk 1]
```diff
@@ -56,24 +75,39 @@ export class GenerateReleaseNotes implements ICommand {
...
+            }
 
-        try {
-            this.toHash = await repo.commitExists(this.toHash);
-            Global.isVerbose() && console.log(`to commit has hash ${this.toHash}`);
-        } catch {
-            throw new Error(`Commit does not exist: ${this.toHash}`);
+            try {
+                this.toHash = await this.repo.commitExists(this.toHash);
+                Global.isVerbose() && console.log(`to commit has hash ${this.toHash}`);
```
Comment (Kaltenbach) [inline src/commands/release-notes/GenerateReleaseNotes.ts:103]:
```suggestion
                Global.isVerbose() && console.log(`--to commit has hash ${this.toHash}`);
```

File: src/commands/release-notes/GenerateReleaseNotes.ts [hunk 2]
```diff
@@ -146,21 +180,56 @@ export class GenerateReleaseNotes implements ICommand {
...
 
-    private async generateChangelog(file: ReleaseNotesMessagesFile, explicits: ReleaseNotesMessagesFile | null, log: IGitLogEntry[]): Promise<void> {
-        const changelog = [`# Changelog ${new Date().toDateString()}`, ''];
-
-        changelog.push('');
-        changelog.push(`_Commit range: ${this.fromHash} - ${this.toHash}_`);
-        changelog.push('');
-
+    private generateChangelog(file: ReleaseNotesMessagesFile, explicits: ReleaseNotesMessagesFile | null, log: IGitLogEntry[]): void {
+        this.changelog = [`_Changelog created on ${new Date().toDateString()}_`];
```
Comment (Kaltenbach) [inline src/commands/release-notes/GenerateReleaseNotes.ts:184]: Please check format for backwards-compatibility with the html we publish to central.
[release notes example in central](https://central.collaboration-factory.de/pages/yglclatbg0lps2xgqbfurr778/22.4?highlightedEntityUid=page%2Fyglclatbg0lps2xgqbfurr778),
See also [the script in the cplace orb](https://github.com/collaborationFactory/cplace-circleci-orb/blob/master/src/scripts/publish_release_notes.sh).

**Review by Kaltenbach — CHANGES_REQUESTED** [review]
Please also move src/test to the new test folder 😄

**Comment by stefancplace** [comment]
compared CHANGELOG.md of @cplace/cli@0.19.12 with this PR

image

**Review by slaven3kopic — APPROVED** [review]
Nice

**Review by Kaltenbach — APPROVED** [review]
Fine for me.

**Review by Kaltenbach — APPROVED** [review]

**Comment by Kaltenbach** [comment]
One thing about the test folder refactoring:
Bildschirmfoto 2022-09-29 um 09 33 48

There are still the two test folders. Is this by intention @stefancplace? Else this should be fixed in a further PR to not block this one. 😄

**Comment by stefancplace** [comment]
I don`t have the folder locally, is there any files in there? @Kaltenbach 
image

**Comment by Kaltenbach** [comment]
Strange, why... the screenshot if from the PR's website...
Bildschirmfoto 2022-09-29 um 09 51 16

**Comment by stefancplace** [comment]
I understand that the picture indicates that `src/test/flow/ReleaseNumber.test.ts` and  `src/test/helpers/idea.ts` have been deleted and are now under `test/flow/ReleaseNumber.test.ts` and `test/helpers/idea.ts` .

**Comment by Kaltenbach** [comment]
> I understand that 

right 🙈 
sorry 😂

**Pull request merged**